### PR TITLE
Add fuzzing targets for integration into OSS-Fuzz

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

This PR adds a few fuzzing targets that can be used with a generic fuzzing engine like `libFuzzer`, with the goal of being integrated into the OSS-Fuzz[1] project for continuous fuzzing.
By default the targets are linked against a standalone target, `standalone_fuzz_target_runner.cc`, which does no fuzzing but makes it easy to check a given test file, or just that the fuzzing targets can be built.
This behaviour can be changed via the `LIB_FUZZING_ENGINE`, for example by setting it to `-lFuzzer` and building with `-fsanitize=fuzzer-no-link` in `CFLAGS` and `CXXFLAGS`.
It makes a lot more sense to enable the `ASAN` and `UBSAN` sanitizers via `--enable-asan` and `--enable-ubsan` options to the `configure`, or using the appropriate flags, when actually fuzzing the targets. 

The current targets cover:
- the auth, dnsdist and rec packet caches (`fuzz_target_packetcache` and `fuzz_target_dnsdistcache`) ;
- `MOADNSParser` ;
- `ZoneParserTNG`.

I'm a bit unsure about the way the targets are currently built, maybe it would make sense to move them to a separate directory. Ideas welcome.

The missing parts are:
- ~~some documentation on how that works~~ ;
- ~~some corpus (DNS queries, DNS zone files)~~ ;
- submitting a PR to oss-fuzz once this PR has been merged.

[1]: https://github.com/google/oss-fuzz

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
A few DNS queries and response, as well as a few zone files (I will actually be reusing the ones from regression-tests/zones/ in the OSS-Fuzz env) have been added to have basic corpora. A small README has been added too.

**Comment by rgacogne** [comment]
Looks ready for a review!

File: fuzzing/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+by setting it to -lFuzzer, building with clang by setting CC=clang CXX=clang++
+before running the configure and adding '-fsanitize=fuzzer-no-link' to CFLAGS
+and CXXFLAGS. Doing so instructs the compiler to instrument the code for
+efficient fuzzing but not to link directly with -lFuzzer, which would make
+the compilation tests done during the configure phase fail.
+
+Sanitizers
+----------
+
+In order to catch the maximum of issues during fuzzing, it makes sene to
```
Comment (ekohl) [inline fuzzing/README.md]: typo: sene -> sense

Comment (rgacogne) [reply fuzzing/README.md]: Fixed, thanks!

**Comment by rgacogne** [comment]
Rebased to fix a conflict.

**Comment by Habbie** [comment]
Builder testing at https://builder.powerdns.com/#/buildrequests/111091?redirect_to_build=true

**Comment by rgacogne** [comment]
> Builder testing at https://builder.powerdns.com/#/buildrequests/111091?redirect_to_build=true

Looks successful!

**Pull request merged**